clusterloader2/pkg/measurement/common/generic_query_measurement.go

```diff
@@ -40,26 +45,61 @@ func init() {
 type genericQueryGatherer struct {
 	metricName    string
 	metricVersion string
-	rawQuery      string
+	queries       []genericQuery
+	unit          string
+}
+
+type genericQuery struct {
+	name         string
+	query        string
+	hasThreshold bool
+	threshold    float64
 }
 
 func (g *genericQueryGatherer) Configure(config *measurement.Config) error {
-	query, err := util.GetString(config.Params, "rawQuery")
+	metricName, err := util.GetString(config.Params, "metricName")
 	if err != nil {
 		return err
 	}
-	metricName, err := util.GetString(config.Params, "metricName")
+
+	metricVersion, err := util.GetString(config.Params, "metricVersion")
 	if err != nil {
 		return err
 	}
-	metricVersion, err := util.GetString(config.Params, "metricVersion")
+
+	unit, err := util.GetString(config.Params, "unit")
 	if err != nil {
 		return err
 	}
 
-	g.rawQuery = query
+	queries, err := util.GetMapArray(config.Params, "queries")
+	if err != nil {
+		return err
+	}
+	var genericQueries []genericQuery
+	for _, q := range queries {
+		query, err := util.GetString(q, "query")
+		if err != nil {
+			return err
+		}
+		name, err := util.GetString(q, "name")
+		if err != nil {
+			return err
+		}
+		hasThreshold := false
+		threshold, err := util.GetFloat64(q, "threshold")
+		if err != nil {
```

Comment:
I think we should distinguish "not found" error (ignore) and other errors (fail). There is a function in util package to check if error type is not found.